activerecord/lib/active_record/connection_handling.rb

```diff
@@ -2,9 +2,6 @@
 
 module ActiveRecord
   module ConnectionHandling
-    RAILS_ENV   = -> { (Rails.env if defined?(Rails.env)) || ENV["RAILS_ENV"].presence || ENV["RACK_ENV"].presence }
-    DEFAULT_ENV = -> { RAILS_ENV.call || "default_env" }
```

Comment:
Since those constants appear in the docs we should deprecate them first and then remove
https://edgeapi.rubyonrails.org/classes/ActiveRecord/ConnectionHandling.html

At first glance, I would probably expect that ActiveRecord::Base.current_environment returns ActiveRecord::InternalMetadata[:environment] value.

Reply:
AFAIK these docs are autogenerated so removing the constants should be enough to remove them once the docs are re-generated, or am I missing something? 

As for `ActiveRecord::InternalMetadata[:environment]` I believe it should be `ActiveRecord::Base.current_environment` that sets that value and not the other way around, similar to what we do today https://github.com/rails/rails/blob/ccb30615a3b6d5fd78b94fe44155310854241f6a/activerecord/lib/active_record/railties/databases.rake#L9

because `InternalMetadata[:environment]` stores the environment for when a migration was run,  when the environment doesn't exists we'd like to default to `default_env`, I also wonder if there are cases where we have not run any migrations and the `:environment` is not set but the `Rails.env` is set.

Reply:
> AFAIK these docs are autogenerated so removing the constants should be enough to remove them once the docs are re-generated, or am I missing something?

Those constants have been in our docs for a long time([since Rails 4.1 as I see](https://api.rubyonrails.org/v4.1/classes/ActiveRecord/ConnectionHandling.html)), so they are being considered as public API, that means that developers could rely on it in their apps/libs and removing those constants without a deprecation would cause breaking changes.

Reply:
We know the our api docs are autogenerated, so unless explicitly opted out by the `# :nodoc:`, it generally be considered as public API even if missing `# :nodoc:` is a legacy reason.

Reply:
cool! I was just wondering if we had to explicitly delete some docs, but seems we are good. 

Totally agree, how do we go about doing a deprecation in rails, is it just an entry in the changelog or is there more we need to do?

Reply:
Assign the constant to a `ActiveSupport::Deprecation::DeprecatedObjectProxy`